### PR TITLE
[test] add Btree test and bug fix

### PR DESCRIPTION
**Pull request by JIACHENG135**

### CODE REVIEW

File: trees/b_tree.py [hunk 1]
```diff
@@ -43,7 +43,7 @@ def split_child(self, x, i):
         # if y is not a leaf, we reassign y's children to y & z
         if not y.leaf:
             z.children = y.children[t: 2 * t]
-            y.children = y.children[0: t - 1]
+            y.children = y.children[0: t]
```
Comment (JIACHENG135) [inline trees/b_tree.py:52]: Add this to pass `test_huge_insert`

**Comment by JIACHENG135** [comment]
`test_huge_insert_and_delete` is still failing. Can you help have a look?

**Comment by JIACHENG135** [comment]
`pytest --log-cli-level=DEBUG ./trees/b_tree_test.py ` to trigger the test

**Comment by msambol** [comment]
Will take a look in the morning 👍🏼

**Comment by JIACHENG135** [comment]
Fixed the bugs and passed all the tests.

**Comment by msambol** [comment]
@JIACHENG135 Hey! I appreciate you digging in. This is a fairly big shift in logic and introduces recursion. I'd like to keep the code as close as possible to the videos. Is there a bug you found that can be fixed with less logic change?

**Comment by JIACHENG135** [comment]
Hey, this is not a huge change since before bug fix the recursive function is already there(which is align with your great video content). I just added the return value for `delete_predecessor` and `delete_successor`since `delete_internal_node` does need that right?

**Comment by JIACHENG135** [comment]
https://github.com/JIACHENG135/Btree

**Comment by msambol** [comment]
> https://github.com/JIACHENG135/Btree

This looks great!

**Pull request closed without merging**